packages/node_modules/@webex/internal-plugin-services/src/services.js

```diff
@@ -110,149 +65,213 @@ const Services = WebexPlugin.extend({
+   * an object that contains the atlas user details if available.
+   * @param {string} email
+   * @returns {Promise<object>} - { activated: boolean, details: string, user: object }
    */
-  waitUntilReady() {
-    return new Promise((resolve) => {
-      if (this.status.ready) {
-        resolve();
-      }
+  validateUser({email, reqId = 'WEBCLIENT'}) {
+    if (!email) {
+      return Promise.reject(new Error('`email` is required'));
+    }
 
-      this.once('ready', () => resolve());
-    });
+    const {canAuthorize, canRefresh} = this.webex.credentials;
+
+    // This conditional statement exists as a logical
+    // fallback in the event that the user is already
+    // authorized via a user token.
+    if (canAuthorize && !canRefresh) {
+      return this.updateServices()
+        .then(() => Promise.resolve({
+          activated: true,
+          details: 'user is authorized via user token',
+          user: {}
+        }));
+    }
+
+    // Scoped function for sending the atlas
+    // activation request.
+    const sendUserActivation = (token) =>
+      this.request({
+        service: 'atlas',
+        resource: 'users/activations',
+        method: 'POST',
+        headers: {
+          accept: 'application/json',
+          authorization: token.toString(),
+          'x-prelogin-userid': undefined
+        },
+        body: {email, reqId},
+        shouldRefreshAccessToken: false
+      });
+
+    // This conditional statement exists as a logical
+    // fallback in the event that the user is already authorized
+    // via a client token.
+    if (canAuthorize && canRefresh) {
+      return this.updateServices()
+        .then(() => this.webex.credentials.getClientToken())
+        .then((token) => sendUserActivation(token.toString()))
+        .then(({body}) => Promise.resolve({
+          activated: true,
+          details: 'user is authorized via client token',
+          user: body
+        }));
+    }
+
+    // Begin unauth user signin process.
+    /* eslint-disable camelcase */
+    const {client_id, client_secret} = this.webex.credentials.config;
+
+    if (!client_id || !client_secret) {
+      return Promise.reject(
+        new Error('`client_id` or `client_secret` is not defined')
+      );
+    }
+    /* eslint-enable camelcase */
+
+    return this.collectPreauthCatalog(email)
+      .then(() => this.webex.credentials.getClientToken({
+        // Core doesn't utilize the service plugin by default.
+        // Inject the required uri via string literal.
+        uri: `${this.get('idbroker', true, 'preauth')}idb/oauth2/v1/access_token`,
+        scope: 'webexsquare:admin webexsquare:get_conversation Identity:SCIM'
+      }))
+      .then(
+        (token) => this.collectSigninCatalog(email, token.toString())
+          // If signin catalog returns 2xx
+          .then(() => sendUserActivation(token)
+            .then(({body}) => Promise.resolve({
```

Comment:
Have this `then` statement attach to the `then` statement on line 207 instead of continuing to indent like this.